cmd/juju/commands/deploy.go

```diff
@@ -334,12 +335,23 @@ func (c *DeployCommand) deployCharmOrBundle(ctx *cmd.Context, client *api.Client
 	// Handle a charm.
 	curl, err := addCharmFromURL(client, charmOrBundleURL, repo, csClient)
 	if err != nil {
+		if error, ok := err.(*errors.Err); ok {
```

Comment:
This doesn't seem quite right. I think the thing closest to the charm deployment should look for an InteractionError; if it finds one, check the the code of the error inside it (using interface{ErrorCode() httpbakery.ErrorCode} and if it's a terms-required code (use string comparison for the code, or import the error constant from somewhere), return a TermsRequiredError holding the relevant details.

The TermsRequiredError.Error method can return the "Declined..." string, and here you'd only need to strip off the annotation if you want to. e.g.

    if err, ok := errors.Cause(err).(*TermsRequiredError); ok {
        return err
    }